File: SourceAFIS/General/Size.cs
using System;

namespace SourceAFIS.General
{
    public struct Size
    {
        public int Width;
        public int Height;

        public Size(int width, int height)
        {
            Width = width;
            Height = height;
        }

        public Size(Point point)
        {
            Width = point.X;
            Height = point.Y;
        }

        public override bool Equals(object other)
        {
            return other is Size && this == (Size)other;
        }

        public override int GetHashCode()
        {
            return Width.GetHashCode() ^ Height.GetHashCode();
        }

        public static bool operator ==(Size left, Size right)
        {
            return left.Width == right.Width && left.Height == right.Height;
        }

        public static bool operator !=(Size left, Size right)
        {
            return !(left == right);
        }
    }
}

File: java/sourceafis/src/sourceafis/general/Size.java
package sourceafis.general;

public class Size {
	public int Width;
	public int Height;

	public Size(int width, int height) {
		Width = width;
		Height = height;
	}

	public Size(Point point) {
		Width = point.X;
		Height = point.Y;
	}

	public boolean equals(Object other) {

		return other instanceof Size && this == other;

		/*
		 * if(!(other instanceof Size)) return false; Size s=(Size)other;
		 * 
		 * return this.Width == s.Width && this.Height == s.Height;
		 */
	}

	public int hashCode() {

		return Width ^ Height;
	}

	public static boolean isEqual(Size left, Size right) {
		return left.Width == right.Width && left.Height == right.Height;
	}

	public static boolean isNotEqual(Size left, Size right) {
		return !isEqual(left, right);
	}

}
